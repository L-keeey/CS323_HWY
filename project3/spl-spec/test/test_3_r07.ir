FUNCTION gcd1 :
PARAM v1
PARAM v2
t3 := v1
t7 := v1
t8 := v2
t5 := t7 / t8
t6 := v2
t4 := t5 * t6
t2 := t3 - t4
v4 := t2
t1 := v4
LABEL label1 :
t9 := v4
t10 := #0
IF t9 != t10 GOTO label2
GOTO label3
LABEL label2 :
t12 := v2
v1 := t12
t11 := v1
t14 := v4
v2 := t14
t13 := v2
t17 := v1
t21 := v1
t22 := v2
t19 := t21 / t22
t20 := v2
t18 := t19 * t20
t16 := t17 - t18
v4 := t16
t15 := v4
GOTO label1
LABEL label3 :
t24 := v2
v3 := t24
t23 := v3
t25 := v3
RETURN t25
FUNCTION gcd2 :
PARAM v5
PARAM v6
t26 := #0
v7 := t26
t27 := v5
t28 := v6
IF t27 > t28 GOTO label4
GOTO label5
LABEL label4 :
t31 := v5
t35 := v5
t36 := v6
t33 := t35 / t36
t34 := v6
t32 := t33 * t34
t30 := t31 - t32
v7 := t30
t29 := v7
t37 := v7
t38 := #0
IF t37 == t38 GOTO label7
GOTO label8
LABEL label7 :
t39 := v6
RETURN t39
GOTO label9
LABEL label8 :
ARG v7
ARG v6
t40 := CALL gcd2
RETURN t40
LABEL label9 :
GOTO label6
LABEL label5 :
t43 := v6
t47 := v6
t48 := v5
t45 := t47 / t48
t46 := v5
t44 := t45 * t46
t42 := t43 - t44
v7 := t42
t41 := v7
t49 := v7
t50 := #0
IF t49 == t50 GOTO label10
GOTO label11
LABEL label10 :
t51 := v5
RETURN t51
GOTO label12
LABEL label11 :
ARG v7
ARG v5
t52 := CALL gcd2
RETURN t52
LABEL label12 :
LABEL label6 :
FUNCTION main :
READ t53
v8 := t53
READ t54
v9 := t54
ARG v9
ARG v8
t55 := CALL gcd1
WRITE t55
ARG v9
ARG v8
t56 := CALL gcd2
WRITE t56
t57 := #0
RETURN t57
