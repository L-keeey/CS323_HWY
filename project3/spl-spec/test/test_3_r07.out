Error type 2 at Line 22: function should be invoked with a definition.
Error type 2 at Line 29: function should be invoked with a definition.
Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: gcd1
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: a
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: b
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: result
            SEMI
          DefList (4)
            Def (4)
              Specifier (4)
                TYPE: int
              DecList (4)
                Dec (4)
                  VarDec (4)
                    ID: temp
              SEMI
        StmtList (5)
          Stmt (5)
            Exp (5)
              Exp (5)
                ID: temp
              ASSIGN
              Exp (5)
                Exp (5)
                  ID: a
                MINUS
                Exp (5)
                  Exp (5)
                    Exp (5)
                      ID: a
                    DIV
                    Exp (5)
                      ID: b
                  MUL
                  Exp (5)
                    ID: b
            SEMI
          StmtList (6)
            Stmt (6)
              WHILE
              LP
              Exp (6)
                Exp (6)
                  ID: temp
                NE
                Exp (6)
                  INT: 0
              RP
              Stmt (6)
                CompSt (6)
                  LC
                  StmtList (7)
                    Stmt (7)
                      Exp (7)
                        Exp (7)
                          ID: a
                        ASSIGN
                        Exp (7)
                          ID: b
                      SEMI
                    StmtList (8)
                      Stmt (8)
                        Exp (8)
                          Exp (8)
                            ID: b
                          ASSIGN
                          Exp (8)
                            ID: temp
                        SEMI
                      StmtList (9)
                        Stmt (9)
                          Exp (9)
                            Exp (9)
                              ID: temp
                            ASSIGN
                            Exp (9)
                              Exp (9)
                                ID: a
                              MINUS
                              Exp (9)
                                Exp (9)
                                  Exp (9)
                                    ID: a
                                  DIV
                                  Exp (9)
                                    ID: b
                                MUL
                                Exp (9)
                                  ID: b
                          SEMI
                  RC
            StmtList (11)
              Stmt (11)
                Exp (11)
                  Exp (11)
                    ID: result
                  ASSIGN
                  Exp (11)
                    ID: b
                SEMI
              StmtList (12)
                Stmt (12)
                  RETURN
                  Exp (12)
                    ID: result
                  SEMI
        RC
    ExtDefList (14)
      ExtDef (14)
        Specifier (14)
          TYPE: int
        FunDec (14)
          ID: gcd2
          LP
          VarList (14)
            ParamDec (14)
              Specifier (14)
                TYPE: int
              VarDec (14)
                ID: x
            COMMA
            VarList (14)
              ParamDec (14)
                Specifier (14)
                  TYPE: int
                VarDec (14)
                  ID: y
          RP
        CompSt (15)
          LC
          DefList (16)
            Def (16)
              Specifier (16)
                TYPE: int
              DecList (16)
                Dec (16)
                  VarDec (16)
                    ID: remainder
                  ASSIGN
                  Exp (16)
                    INT: 0
              SEMI
          StmtList (17)
            Stmt (17)
              IF
              LP
              Exp (17)
                Exp (17)
                  ID: x
                GT
                Exp (17)
                  ID: y
              RP
              Stmt (17)
                CompSt (17)
                  LC
                  StmtList (18)
                    Stmt (18)
                      Exp (18)
                        Exp (18)
                          ID: remainder
                        ASSIGN
                        Exp (18)
                          Exp (18)
                            ID: x
                          MINUS
                          Exp (18)
                            Exp (18)
                              Exp (18)
                                ID: x
                              DIV
                              Exp (18)
                                ID: y
                            MUL
                            Exp (18)
                              ID: y
                      SEMI
                    StmtList (19)
                      Stmt (19)
                        IF
                        LP
                        Exp (19)
                          Exp (19)
                            ID: remainder
                          EQ
                          Exp (19)
                            INT: 0
                        RP
                        Stmt (20)
                          RETURN
                          Exp (20)
                            ID: y
                          SEMI
                        ELSE
                        Stmt (22)
                          RETURN
                          Exp (22)
                            ID: gcd2
                            LP
                            Args (22)
                              Exp (22)
                                ID: y
                              COMMA
                              Args (22)
                                Exp (22)
                                  ID: remainder
                            RP
                          SEMI
                  RC
              ELSE
              Stmt (24)
                CompSt (24)
                  LC
                  StmtList (25)
                    Stmt (25)
                      Exp (25)
                        Exp (25)
                          ID: remainder
                        ASSIGN
                        Exp (25)
                          Exp (25)
                            ID: y
                          MINUS
                          Exp (25)
                            Exp (25)
                              Exp (25)
                                ID: y
                              DIV
                              Exp (25)
                                ID: x
                            MUL
                            Exp (25)
                              ID: x
                      SEMI
                    StmtList (26)
                      Stmt (26)
                        IF
                        LP
                        Exp (26)
                          Exp (26)
                            ID: remainder
                          EQ
                          Exp (26)
                            INT: 0
                        RP
                        Stmt (27)
                          RETURN
                          Exp (27)
                            ID: x
                          SEMI
                        ELSE
                        Stmt (29)
                          RETURN
                          Exp (29)
                            ID: gcd2
                            LP
                            Args (29)
                              Exp (29)
                                ID: x
                              COMMA
                              Args (29)
                                Exp (29)
                                  ID: remainder
                            RP
                          SEMI
                  RC
          RC
      ExtDefList (33)
        ExtDef (33)
          Specifier (33)
            TYPE: int
          FunDec (33)
            ID: main
            LP
            RP
          CompSt (34)
            LC
            DefList (35)
              Def (35)
                Specifier (35)
                  TYPE: int
                DecList (35)
                  Dec (35)
                    VarDec (35)
                      ID: m
                    ASSIGN
                    Exp (35)
                      ID: read
                      LP
                      RP
                SEMI
              DefList (36)
                Def (36)
                  Specifier (36)
                    TYPE: int
                  DecList (36)
                    Dec (36)
                      VarDec (36)
                        ID: n
                      ASSIGN
                      Exp (36)
                        ID: read
                        LP
                        RP
                  SEMI
            StmtList (37)
              Stmt (37)
                Exp (37)
                  ID: write
                  LP
                  Args (37)
                    Exp (37)
                      ID: gcd1
                      LP
                      Args (37)
                        Exp (37)
                          ID: m
                        COMMA
                        Args (37)
                          Exp (37)
                            ID: n
                      RP
                  RP
                SEMI
              StmtList (38)
                Stmt (38)
                  Exp (38)
                    ID: write
                    LP
                    Args (38)
                      Exp (38)
                        ID: gcd2
                        LP
                        Args (38)
                          Exp (38)
                            ID: m
                          COMMA
                          Args (38)
                            Exp (38)
                              ID: n
                        RP
                    RP
                  SEMI
                StmtList (39)
                  Stmt (39)
                    RETURN
                    Exp (39)
                      INT: 0
                    SEMI
            RC
Program
ExtDefList
ExtDef
FunDec
FUNCTION gcd1 :
VarList
ParamDec
VarDec
PARAM v1
VarList
ParamDec
VarDec
PARAM v2
ComSt
DefList
Def
DecList
Dec
VarDec
DefList
Def
DecList
Dec
VarDec
DefList
StmtList
Stmt
*******
Exp
ASSIGN
Exp
*******
Exp
assign
v4
Exp
Exp
t3 := v1
Exp
Exp
Exp
t7 := v1
Exp
t8 := v2
t5 := t7 / t8
Exp
t6 := v2
t4 := t5 * t6
t2 := t3 - t4
v4 := t2
t1 := v4
StmtList
Stmt
LABEL label1 :
cond_Exp
3
Exp
t9 := v4
Exp
const
INT
0
t10 := #0
IF t9 != t10 GOTO label2
GOTO label3
LABEL label2 :
Stmt
ComSt
DefList
StmtList
Stmt
*******
Exp
ASSIGN
Exp
*******
Exp
assign
v1
Exp
t12 := v2
v1 := t12
t11 := v1
StmtList
Stmt
*******
Exp
ASSIGN
Exp
*******
Exp
assign
v2
Exp
t14 := v4
v2 := t14
t13 := v2
StmtList
Stmt
*******
Exp
ASSIGN
Exp
*******
Exp
assign
v4
Exp
Exp
t17 := v1
Exp
Exp
Exp
t21 := v1
Exp
t22 := v2
t19 := t21 / t22
Exp
t20 := v2
t18 := t19 * t20
t16 := t17 - t18
v4 := t16
t15 := v4
StmtList
GOTO label1
LABEL label3 :
StmtList
Stmt
*******
Exp
ASSIGN
Exp
*******
Exp
assign
v3
Exp
t24 := v2
v3 := t24
t23 := v3
StmtList
Stmt
Exp
t25 := v3
RETURN t25
StmtList
ExtDefList
ExtDef
FunDec
FUNCTION gcd2 :
VarList
ParamDec
VarDec
PARAM v5
VarList
ParamDec
VarDec
PARAM v6
ComSt
DefList
Def
DecList
Dec
Exp
const
INT
0
t26 := #0
v7 := t26
DefList
StmtList
Stmt
cond_Exp
3
Exp
t27 := v5
Exp
t28 := v6
IF t27 > t28 GOTO label4
GOTO label5
LABEL label4 :
Stmt
ComSt
DefList
StmtList
Stmt
*******
Exp
ASSIGN
Exp
*******
Exp
assign

Exp
Exp
t31 := v5
Exp
Exp
Exp
t35 := v5
Exp
t36 := v6
t33 := t35 / t36
Exp
t34 := v6
t32 := t33 * t34
t30 := t31 - t32
